### PR TITLE
Refactor Featured component

### PR DESCRIPTION
**Pull request by NicolasMahe**
### Description

Refactor Featured component in order to be able to flatten `sales`, `ownerships` from `asset`.
The component `FeaturedToken` (previously `TokenHeader`) is now doing the query itself to be able to flatten the query.

Remove now deprecated `totalAvailableQuantitySum`. Use the `sales.availableQuantity` to calculate it. I completely remove the `limit: 100` that doesn't have pagination anyway.

@ismailToyran could you check if I didn't break anything please.

### Checklist

- [x] Base branch of the PR is `dev`

### CODE REVIEW

File: components/Token/Header.tsx [hunk 1]
```diff
@@ -1,104 +0,0 @@
-import {
```
Comment (NicolasMahe) [inline components/Token/Header.tsx:1]: This file is moved to HomeSection/FeaturedToken as it was only use their

File: components/HomeSection/FeaturedToken.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+  const fetchFeaturedTokenQuery = useFetchFeaturedTokenQuery({
+    variables: {
+      chainId,
+      collectionAddress,
+      tokenId,
+      now: date,
+      address: address || '',
+    },
+  })
```
Comment (NicolasMahe) [inline components/HomeSection/FeaturedToken.tsx:45]: This component is now fetch the asset itself.

File: components/HomeSection/FeaturedToken.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+  const media = useDetectAssetMedia(asset)
+
+  const refresh = useCallback(async () => {
+    await fetchFeaturedTokenQuery.refetch({
+      now: new Date(),
+    })
+  }, [fetchFeaturedTokenQuery])
+
+  useCart({ onCheckout: fetchFeaturedTokenQuery.refetch })
```
Comment (NicolasMahe) [inline components/HomeSection/FeaturedToken.tsx:60]: The refetch `onCheckout` was missing from `Featured.tsx`

**Review by ismailToyran — APPROVED** [review]
Looking good from my side with the comments present on the PR 👍

**Pull request merged**